### PR TITLE
fix syntax of altermanager::receivers

### LINKED ISSUE #539: alertmanager receivers incorrect syntax
## How to reproduce (e.g Puppet code you use)

In Debian:

```
  class { 'prometheus::alertmanager':
    package_ensure => 'present',
    service_ensure => 'running',
    install_method => 'package',
    init_style     => 'none',
    user           => 'prometheus',
    group          => 'prometheus',
    package_name   => 'prometheus-alertmanager',
    service_name   => 'prometheus-alertmanager',
    # disable clustering
    extra_options  => '--cluster.listen-address=',
}
```

But I suspect a simple:

```
include prometheus::alertmanager
```

would also trigger this.

## What are you seeing

```
Mar 31 18:16:43 hetzner-nbg1-02 prometheus-alertmanager[9316]: level=info ts=2021-03-31T18:16:43.376918397Z caller=main.go:322 msg="Loading configuration file" file=/etc/alertmanager/alertmanager.yaml
Mar 31 18:16:43 hetzner-nbg1-02 prometheus-alertmanager[9316]: level=error ts=2021-03-31T18:16:43.377199383Z caller=main.go:325 msg="Loading configuration file failed" file=/etc/alertmanager/alertmanager.yaml err="missing name in receiver"
Mar 31 18:16:43 hetzner-nbg1-02 systemd[1]: prometheus-alertmanager.service: Main process exited, code=exited, status=1/FAILURE
```

## What behaviour did you expect instead

If I add this to the class, it works:

```
    receivers      => [
      {
        'name'          => 'Admin',
        'email_configs' => [
          'to'          => 'root@localhost',
        ],
      }
    ],
```

I suspect the syntax of the YAML file is incorrect.

### PR DESCRIPTION
**Pull request by TheMeier**
fixes #539

### CODE REVIEW

**Comment by TheMeier** [comment]
Existing test is sufficient, with wrong config the acceptance test fails with:

```
  Error: Execution of '/usr/local/bin/amtool check-config --alertmanager.url='' /etc/alertmanager/alertmanager.yaml20240511-3098-1nt8gcc' returned 1: Checking '/etc/alertmanager/alertmanager.yaml20240511-3098-1nt8gcc'  FAILED: missing name in receiver
```

**Comment by TheMeier** [comment]
Archlinux tests are addressed in https://github.com/voxpupuli/puppet-prometheus/pull/719

File: data/defaults.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ prometheus::alertmanager::time_intervals: []
 prometheus::alertmanager::storage_path: '/var/lib/alertmanager'
 prometheus::alertmanager::templates: [ "%{lookup('prometheus::alertmanager::config_dir')}/*.tmpl" ]
 prometheus::alertmanager::user: 'alertmanager'
-prometheus::alertmanager::version: '0.21.0'
+prometheus::alertmanager::version: '0.27.0'
```
Comment (bastelfreak) [inline data/defaults.yaml]: if we update it, the PR title should reflect this as well.

File: spec/acceptance/alertmanager_spec.rb [hunk 2]
```diff
@@ -6,21 +6,7 @@
...
-                email_configs => [
-                  send_resolved => true,
-                  to => 'test@localhost.localdomain',
-                  from => 'prometheus@localhost.localdomain',
-                  smarthost => '127.0.0.1:25',
-                  require_tls => false
-                ]
-              }
-            ]
+            version => '0.27.0',
```
Comment (bastelfreak) [inline spec/acceptance/alertmanager_spec.rb]: you don't need to set this version when it's in hiera?

File: data/defaults.yaml (hunk 1 above)
Comment (TheMeier) [reply data/defaults.yaml]: removed the version change, will update seperatly

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**